### ISSUE TITLE
`zip_paper` creates incomplete submissions

### ISSUE THREAD

**Issue by s-u** [rjournal/rjtools#69]
`zip_paper` is very misleading as it doesn't create valid submissions unless the files follow very specific naming conventions which are not documented. It doesn't include any dependencies used in the sources such as images unless they happen to be in a sub-directory with `name` in it (since all it does is to use `name` as regexp). This causes issues with common conventions such as figures in the `figures` or `images` folder. Also the ZIP file does not include files necessary for the LaTeX rendering.

- the requirements should be documented such that it is clear where the sources have to reside for `zip_paper` to work. Ideally it would be nice if it could find the location of included files such as images automatically
- the ZIP file should include both html and LaTeX artifacts (possibly creating them if not present) such that the submissions can processed directly with `pdflatex`
- it would be helpful to check that the resulting ZIP file is complete (i.e. test that both html and tex rendering works on the result)

**Comment by huizezhang-sherry** [comment]
Hi Simon, 

Additional files like figures or data can be included in the zip with the argument `others`. For example, `rjtools::zip_paper(NAME, others = c("figs", "data/xxx.rda"))` will include all the figures under the figs folder and the xxx.rda data in the zip. I think the argument can have a better name, say, `inlcude` and better documentation. 

Another option is to auto-detect commonly used image folder name (img, figures, images, etc) and ask users whether they would like to include them.

**Comment by s-u** [comment]
Yes, definitely better documentation should be the first step, in particular since this is supposed to be function to prepare submissions. I'm still a bit uneasy with the approach of using substring search to pick files - I think it would be better to define some guidelines on the structure instead - like R packages where we have well-defined directory names and structure for contents that is to be submitted.
